app/uk/gov/hmrc/tariffclassificationfrontend/models/Case.scala

```diff
@@ -36,5 +37,20 @@ case class Case
   keywords: Set[String] = Set.empty
 ) {
 
-  def rulingHasNotExpired: Boolean = decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) >= 0)
+  private def hasDefinedRuling: Boolean = {
+    decision match {
+      case Some(d) if d.effectiveEndDate.isDefined => true
+      case _ => false
+    }
+  }
+
+  def hasExpiredRuling(implicit appConfig: AppConfig): Boolean = {
+    val currentTime = Instant.now(appConfig.clock)
+    decision.flatMap(_.effectiveEndDate).exists(_.isBefore(currentTime))
+  }
+
+  def hasLiveRuling(implicit appConfig: AppConfig): Boolean = {
+    hasDefinedRuling && !hasExpiredRuling
```

Comment:
I'm not sure you need the `hasDefinedRuling` here, I think `!hasExpiredRuling` does the same check already

Reply:
I do not agree with you.
This means that you return `true` for a case without the `decision` field or without the `effectiveEndDate` field

Reply:
Okay good point